scripts/travis.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# if this build was triggered as cron job, run e2e tests on iOS
```

Comment:
when the build will be triggered as cron job?

Reply:
Right now, I've scheduled it to run this branch as cron daily here - https://travis-ci.org/okta/okta-signin-widget/settings (Scroll down to Cron Jobs section)
If we merge this PR, we can run the master branch instead.

Reply:
Can you explain why we decided to include this coverage **only** via a cron event type?

Moving forward, I'd assume we'd like to ensure that the Widget works properly on supported Android and iOS versions. If we have failing tests for Android or iOS, I would expect to have Travis block the PR from being merged. Is this something we're gradually moving toward? Are tests on these devices simply not working today?

Reply:
The main reason was travis run time increased by almost 50%. The existing tests take around 15 mins. The iOS tests run on 3 device emulators (serially) and take around 15 - 18 mins. After discussing it with @robertjd and @manueltanzi-okta, we decided it's good to start this as a cron job and explore if tests can be run in parallel.
As for Android tests, there are a couple more failing tests in oidc_spec. Initially I wanted to get both iOS + Android in this PR but then I have to add more disable logic in tests. Hence decided against it. Will work on Android in the next sprint.

Reply:
Gotcha - that significantly slows down development time.

The primary reason I'm calling this out is in the past, tests that have failed via CRON jobs have been neglected. The `htmlproofer` tests within D.O.C were a great example of this. 

Another problem with cron tasks is delegating **who** is responsible for resolving the failing tests, as we'd need to track down which commit introduced the failures.

To get around this, what are your thoughts on running the Appium tests within Bacon _instead_ of Travis? When new versions of the Widget are being promoted, we'll run the Appium suite in parallel with the existing tests. That way - we can guarantee that no new versions of the Widget can be released without passing the mobile browser tests.

What do you think?

Reply:
> When new versions of the Widget are being promoted, we'll run the Appium suite in parallel with the existing tests

Reply:
> What about 'when an new commit hit master, run Appium suite'?

I thought about that originally, but then we're forced to react to the possibility of broken tests that have already been merged. Ideally we'll catch errors **before** the topic branch has been merged.

This gives us two options:
1. Run the Appium tests against all topic branches with the side-effect of doubling our CI time (~30min)
2. Run the Appium tests in Bacon to support parallelism. Similar to `bacon-topic-*` prefixes, we can run this against topic or `master`.

I'd advocate for option two, as it keeps our Travis CI time minimal. Plus, I'd assume the primary reason for running the entire browser suite (embedded + mobile) is for major/minor releases or drastic view changes. We'd just need to encourage internal devs to prefix their branches accordingly for changes that _may_ break embedded or mobile browser tests.

> and we might want to set up the testing in an way that developer has option to run those test in topic branch

Somewhat related to the first point - I'd assume we'd want this option to be available for internal contributors using our internal CI to provide some sense of a sanity check. If we wanted all external PRs to run our Appium tests (probably don't want to for security reasons), we can always keep our existing pattern of fetching an external contributor's topic branch and pushing it to a remote branch ourselves.

Reply:
Running on bacon seems like a good approach.
I see that we run unit tests on bacon today for master and bacon-topic* branches. On the topic branch, does it block merges if unit tests fail?

Reply:
It doesn't block merges since merges are handled by Travis. The bacon regex is primarily used internally as validation that we're not breaking any of our core sign-in flows.

As I understand the current flow, when we're making drastic changes or getting ready to release minor/major versions, we run our internal CI suite for a sanity check. There is no _hard_ requirement for this, but oftentimes will get called out in a PR review.